packages/sdk/src/utils.ts

```diff
@@ -72,9 +72,12 @@ export async function loop(fn: (...args: any[]) => any, delay: number, ...args:
   } while (true);
 }
 
-type Call = [string, ...any[]];
-type Calls = Call[];
-export const BatchReadWithErrors = (multicall2: Multicall2) => (contract: Contract) => async (calls: Calls) => {
+export type Call = [string, ...BigNumberish[]];
+export type Calls = Call[];
+export type BatchReadWithErrorsType = ReturnType<ReturnType<typeof BatchReadWithErrors>>;
+export const BatchReadWithErrors = (multicall2: Multicall2) => (contract: Contract) => async <R>(
```

Comment:
Minor note: is there a strong reason to write this function this way? Not saying we should change it since it is pre-existing, but I think the call syntax `BatchReadWithErrors(this.multicall)(this.contract)(calls)` is not very idiomatic for js/ts and can be confusing at the callsite.

Reply:
The only reason really is so you can build multiple batch readers from a single provider and multicall instance. In a more complex scenario, you would instantiate a single level with your multicall contract at the top of your app, and pass that down to specific components which could pass the contract they want to multicell. From there the specific component could use that to make batch calls on that contract in multiple places.

This pattern is specific to the type of client i plan to build, but most likely you would just want to use the multicall class directly.